http-client/src/main/java/io/micronaut/http/client/netty/DefaultHttpClient.java

```diff
@@ -2085,30 +2131,40 @@ private static class CurrentEvent {
         Duration retry;
     }
 
-    private abstract class BaseHttpResponseHandler<O> extends SimpleChannelInboundHandlerInstrumented<HttpObject> {
+    private abstract class BaseHttpResponseListener<O> implements Http1ResponseHandler.ResponseListener {
         final Promise<? super O> responsePromise;
         final io.micronaut.http.HttpRequest<?> parentRequest;
         final io.micronaut.http.HttpRequest<?> finalRequest;
 
-        public BaseHttpResponseHandler(Promise<? super O> responsePromise, io.micronaut.http.HttpRequest<?> parentRequest, io.micronaut.http.HttpRequest<?> finalRequest) {
-            super(false);
+        public BaseHttpResponseListener(Promise<? super O> responsePromise, io.micronaut.http.HttpRequest<?> parentRequest, io.micronaut.http.HttpRequest<?> finalRequest) {
             this.responsePromise = responsePromise;
             this.parentRequest = parentRequest;
             this.finalRequest = finalRequest;
         }
 
-        @Override
-        public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
-            ctx.read();
+        private static void setRedirectHeaders(@Nullable io.micronaut.http.HttpRequest<?> request, MutableHttpRequest<Object> redirectRequest) {
+            if (request != null) {
```

Comment:
I would prefer check for null and `return` to void more nesting

Reply:
don't want to touch because this is old code.